src/components/ManageWallets/ManageWalletsRow.tsx

```diff
@@ -20,36 +24,23 @@ type Props = {
   isOnlyWalletConnected: boolean;
 };
 
-function ManageWalletsRow({
-  walletId,
-  address,
-  chain,
-  userSigninAddress,
-  setErrorMessage,
-  setRemovedAddress,
-  isOnlyWalletConnected,
-}: Props) {
+function ManageWalletsRow({ walletId, address, chain, setErrorMessage, setRemovedAddress }: Props) {
   const removeWallet = useRemoveWallet();
+  const updatePrimaryWallet = useUpdatePrimaryWallet();
   const [isDisconnecting, setIsDisconnecting] = useState(false);
-  const [showTooltip, setShowTooltip] = useState(false);
-
-  const handleMouseEnter = useCallback(() => {
-    setShowTooltip(true);
-  }, []);
-
-  const handleMouseExit = useCallback(() => {
-    setShowTooltip(false);
-  }, []);
+  const [isPending, setIsPending] = useState(false);
 
   const handleDisconnectClick = useCallback(async () => {
     ReactTooltip.hide();
     try {
       setErrorMessage('');
       setIsDisconnecting(true);
+      setIsPending(true);
```

Comment:
The underlying `usePromisifiedMutation` returns `[mutate, isMutating]`. Can we use the underlying flag here instead of creating our own state or is this necessary?

Reply:
ooh I didn't realize. sounds promising

Reply:
uhh no pun intended

Reply:
worked like a charm. updated both the `useUpdatePrimaryWallet` and `useRemoveWallet` hooks so I was able to remove `useState` entirely from this file, thank you ser